apps/mobile/src/screens/Onboarding/OnboardingRecommendedUsers.tsx

```diff
@@ -0,0 +1,201 @@
+import { removeNullValues } from '~/shared/relay/removeNullValues';
+import useFollowAllRecommendedUsers from '~/shared/relay/useFollowAllRecommendedUsers';
+import colors from '~/shared/theme/colors';
+import { noop } from '~/shared/utils/noop';
+
+import { navigateToNotificationUpsellOrHomeScreen } from '../Login/navigateToNotificationUpsellOrHomeScreen';
+
+const HEIGHT_OF_FIXED_ITEMS_ON_SCREEN = 270;
+
+export function OnboardingRecommendedUsers() {
+  const query = useLazyLoadQuery<OnboardingRecommendedUsersQuery>(
+    graphql`
+      query OnboardingRecommendedUsersQuery($usersLast: Int!, $usersBefore: String) {
+        ...OnboardingRecommendedUsersInnerFragment
+      }
+    `,
+    {
+      usersLast: USERS_PER_PAGE,
+      usersBefore: null,
+    }
+  );
+
+  return <OnboardingRecommendedUsersInner queryRef={query} />;
+}
+
+type OnboardingRecommendedUsersInnerProps = {
+  queryRef: OnboardingRecommendedUsersInnerFragment$key;
+};
+
+function OnboardingRecommendedUsersInner({ queryRef }: OnboardingRecommendedUsersInnerProps) {
+  const {
+    data: followingPagination,
+    loadPrevious,
+    hasPrevious,
+  } = usePaginationFragment(
+    graphql`
+      fragment OnboardingRecommendedUsersInnerFragment on Query
+      @refetchable(queryName: "OnboardingRecommendedUsersInnerQuery") {
+        viewer {
+          ... on Viewer {
+            id
+
+            suggestedUsers(last: $usersLast, before: $usersBefore)
+              @connection(key: "OnboardingRecommendedUsersInnerFragment_suggestedUsers") {
+              edges {
+                node {
+                  id
+                  __typename
+                  ...UserFollowListFragment
+                }
+              }
+            }
+          }
+        }
+        ...useFollowAllRecommendedUsersQueryFragment
+        ...UserFollowListQueryFragment
+      }
+    `,
+    queryRef
+  );
+
+  const navigation = useNavigation<LoginStackNavigatorProp>();
+
+  const user = followingPagination?.viewer?.id;
+
+  const suggestedFollowingIds = useMemo(() => {
+    const userIds = [];
+
+    for (const edge of followingPagination.viewer?.suggestedUsers?.edges ?? []) {
+      if (edge?.node?.__typename === 'GalleryUser') {
+        userIds.push(edge?.node.id);
+      }
+    }
+
+    return userIds;
+  }, [followingPagination]);
+
+  const windowDimensions = useWindowDimensions();
+
+  const heightOfUserList = windowDimensions.height - HEIGHT_OF_FIXED_ITEMS_ON_SCREEN;
+
+  const followAllRecommendedUsers = useFollowAllRecommendedUsers({
+    suggestedFollowingIds: suggestedFollowingIds,
+    queryRef: followingPagination,
+  });
+
+  const handleNext = useCallback(async () => {
+    if (!user) return null;
+    await navigateToNotificationUpsellOrHomeScreen(navigation, true);
+  }, [navigation, user]);
+
+  const handleFollowAll = useCallback(async () => {
+    if (!user) return null;
+    followAllRecommendedUsers();
+    setTimeout(async () => await handleNext(), 800);
```

Comment:
Why do we need `setTimeout` here? Are we waiting for the request to finish

If yes, maybe if there is any loading state would be good or can utilize the `updater` or `optimisticResponse` in the relay